src/App.jsx

```diff
@@ -0,0 +1,187 @@
+import { useState, useEffect } from "react";
+import useFetchWeather from "./hooks/useFetchWeather";
+import TodayWeather from "./components/TodayWeather/TodayWeather";
+import LocationSearch from "./components/LocationSearch/LocationSearch";
+import RoundButton from "./components/RoundButton/RoundButton";
+import WeatherCard from "./components/WeatherCard/WeatherCard";
+import HighlightCard from "./components/HighlightCard/HighlightCard";
+import Skeleton from "./components/Skeleton/Skeleton";
+import { WEATHER_TYPES, UNITS } from "./constants";
+import "./App.css";
+
+function App() {
+  const initialLocation = "Firenze";
+  const {
+    currentWeatherData,
+    forecastWeatherData,
+    error,
+    unit,
+    isLoading,
+    setUnit,
+    fetchWeather,
+    handleTemperatureUnit,
+  } = useFetchWeather(initialLocation);
+  const [isLocationSearchOpen, setIsLocationSearchOpen] = useState(false);
+  const [locations, setLocations] = useState([]);
+
+  useEffect(() => {
+    const storedLocations = localStorage.getItem("locations");
+    if (storedLocations) {
+      setLocations(JSON.parse(storedLocations));
+    }
+  }, []);
+
+  useEffect(() => {
+    if (locations.length > 0) {
+      localStorage.setItem("locations", JSON.stringify(locations));
+    }
+  }, [locations]);
+
+  const handleLocationSearch = async (location) => {
+    const isSuccessful = await fetchWeather(location, WEATHER_TYPES.CURRENT);
+    if (isSuccessful && typeof location === "string") {
+      setLocations((locations) => {
+        if (!locations.includes(location)) {
+          return [...locations, location];
+        }
+        return locations;
+      });
+      handleLocationSearchVisibility();
+    }
+    await fetchWeather(location, WEATHER_TYPES.FORECAST);
+  };
```

Comment:
For here, you're using 2 fetchs, one first and then another, let's use the new one that will aggregate both.
Also, instead of `isSuccessful`, let's use `isValidLocation`, so it makes a bit more sense, since you want to add locations only when valid.

Also, prefer to use `prevLocations` to be clear you're merging with the previous ones. Also, you can use more use of the ternary to return one or another, saving `return` and `if`

```suggestion
const handleLocationSearch = async (location) => {
  const isValidLocation = await fetchWeatherForLocation(location);
  
  if (isValidLocation && typeof location === "string") {
    setLocations((prevLocations) =>
      !prevLocations.includes(location)
        ? [...prevLocations, location]
        : prevLocations
    );
    
    handleLocationSearchVisibility();
  }
};
```